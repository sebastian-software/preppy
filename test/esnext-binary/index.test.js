/* global __dirname */
import { readFile } from "fs"
import { resolve } from "path"

import pify from "pify"
import rimraf from "rimraf"

import { preppy } from "../util"

const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)


test("Binary from ESNext", async () => {
  await lazyDelete(resolve(__dirname, "./bin"))

  await preppy({
    cwd: __dirname
  })

  expect(await lazyRead(resolve(__dirname, "bin/mycli.js"), "utf8")).toMatchSnapshot("cli")
})
