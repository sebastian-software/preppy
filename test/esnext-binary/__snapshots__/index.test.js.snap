// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Binary from ESNext: cli 1`] = `
"#!/usr/bin/env node

/*! esnext-binary v1.0.0 by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

var _snakeCase = require('lodash/snakeCase');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e[\\"default\\"] : e; }

var _snakeCase__default = /*#__PURE__*/_interopDefaultLegacy(_snakeCase);

const realWords = \\"Eine dicke Kuh lacht\\";

console.log(\\"Magic:\\", _snakeCase__default(realWords));
//# sourceMappingURL=mycli.js.map
"
`;
